tests/JustSaying.Extensions.DependencyInjection.AwsCore.Tests/ClientFactoryTests.cs

```diff
@@ -31,76 +41,88 @@ public void ShouldReadConfigFromFile()
     [Fact]
     public void ShouldReadFromDefaultProfile()
     {
-        Environment.SetEnvironmentVariable("AWS_PROFILE", null);
-        var bin = AppContext.BaseDirectory;
-        Environment.SetEnvironmentVariable("AWS_CONFIG_FILE", $"{bin}/config");
-        Environment.SetEnvironmentVariable("AWS_SHARED_CREDENTIALS_FILE", $"{bin}/credentials");
+        //Arrange
         var config = new ConfigurationBuilder()
             .AddEnvironmentVariables()
             .Build();
         var serviceCollection = new ServiceCollection();
-        serviceCollection.AddJustSyaingWithAwsConfig(config, (_) => { });
+        serviceCollection.AddJustSayingWithAwsConfig(config, (_) => { });
+
+        //Act
         var provider = serviceCollection.BuildServiceProvider();
         var clientFactory = provider.GetRequiredService<IAwsClientFactory>();
-        var snsClient = clientFactory.GetSnsClient(RegionEndpoint.EUWest1);
+        var snsClient = clientFactory.GetSnsClient(RegionEndpoint.EUWest2);
+
+        //Assert
         Assert.True(string.IsNullOrEmpty(snsClient.Config.ServiceURL), snsClient.Config.ServiceURL);
-        Assert.True(snsClient.Config.RegionEndpoint == RegionEndpoint.EUWest1);
+        Assert.True(snsClient.Config.RegionEndpoint == RegionEndpoint.EUWest2);
         Assert.False(snsClient.Config.UseHttp);
-        Environment.SetEnvironmentVariable("AWS_CONFIG_FILE", null);
-        Environment.SetEnvironmentVariable("AWS_SHARED_CREDENTIALS_FILE", null);
-        Environment.SetEnvironmentVariable("AWS_PROFILE", null);
 
     }
 
     [Fact]
-    public void ShouldReadFromSpecifiedProfile()
+    public void ShouldReadFromConfigValues()
     {
-        var bin = AppContext.BaseDirectory;
-        Environment.SetEnvironmentVariable("AWS_CONFIG_FILE", $"{bin}/config");
-        Environment.SetEnvironmentVariable("AWS_SHARED_CREDENTIALS_FILE", $"{bin}/credentials");
-        Environment.SetEnvironmentVariable("AWS_PROFILE", "local");
+        //Arrange
         var config = new ConfigurationBuilder()
-            .AddEnvironmentVariables()
+            .AddInMemoryCollection(new Dictionary<string, string>
+            {
+                {"AWS:Profile", "local"},
+                {"AWS:ProfilesLocation", _credentialsFile},
+                {"AWS:Region", "us-east-1"},
+                {"AWS:ServiceURL", "http://profile.test/"},
+                {"AWS:UseHttp", "true"}
```

Comment:
Newer C# syntax to initialise dictionaries is slightly less verbose and nicer to read IMHO:

```suggestion
                ["AWS:Profile"] = "local",
                ["AWS:ProfilesLocation"] = _credentialsFile,
                ["AWS:Region"] = "us-east-1",
                ["AWS:ServiceURL"] = "http://profile.test/",
                ["AWS:UseHttp"] = "true",
```